src/sqlfluff/dialects/dialect_greenplum.py

```diff
@@ -167,13 +248,77 @@ class CreateTableStatementSegment(postgres.CreateTableStatementSegment):
                 OneOf(Sequence("PRESERVE", "ROWS"), Sequence("DELETE", "ROWS"), "DROP"),
             ),
             Sequence("TABLESPACE", Ref("TablespaceReferenceSegment")),
+            Ref("DistributedBySegment")
+        ),
+    )
+
+
+class CreateTableAsStatementSegment(postgres.CreateTableAsStatementSegment):
+    """A `CREATE TABLE AS` statement.
+
+    As specified in
+    https://docs.vmware.com/en/VMware-Tanzu-Greenplum/6/greenplum-database/GUID-ref_guide-sql_commands-CREATE_TABLE_AS.html
+    This is overriden from Postgres to add the `DISTRIBUTED` clause.
+    """
+
+    match_grammar = Sequence(
+        "CREATE",
+        OneOf(
             Sequence(
-                "DISTRIBUTED",
-                OneOf(
-                    "RANDOMLY",
-                    "REPLICATED",
-                    Sequence("BY", Bracketed(Ref("ColumnReferenceSegment"))),
+                OneOf("GLOBAL", "LOCAL", optional=True),
+                Ref("TemporaryGrammar"),
+            ),
+            "UNLOGGED",
+            optional=True,
+        ),
+        "TABLE",
+        Ref("IfNotExistsGrammar", optional=True),
+        Ref("TableReferenceSegment"),
+        AnyNumberOf(
+            Sequence(
```

Comment:
You don't need this Sequence(), you can stick the optional=True in Bracketed()

Reply:
Ok - this was copied straight from the postgres dialect file so I left it alone

Reply:
And sorted this.